codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -764,12 +764,12 @@ private InterfaceHttpData findMultipartDisposition() {
                     }
                 } else {
                     for (int i = 1; i < contents.length; i++) {
-                        if (contents[i].toLowerCase().startsWith(HttpHeaderValues.CHARSET.toString())) {
+                        final String charsetHeader = HttpHeaderValues.CHARSET.toString();
+                        if (contents[i].regionMatches(true, 0, charsetHeader, 0, charsetHeader.length())) {
```

Comment:
It may change existing flows. My commit doesn't change any flows, just performance improvement. This part is not covered well with tests so I wouldn't risk here. Maybe later after deeper code review.